docs/current/guides/723462-use-secrets.md

```diff
@@ -0,0 +1,305 @@
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/use-secrets/raw/main.go
+```
+
+In this code listing:
+
+- The client's `SetSecret()` method accepts two parameters - a unique name for the secret, and the secret value - and returns a new `Secret` object
+- The `WithSecretVariable()` method also accepts two parameters - an environment variable name and a `Secret` object. It returns a container with the secret value assigned to the specified environment variable.
+- The environment variable can then be used in subsequent container operations - for example, in a command-line `curl` request, as shown above.
+
+</TabItem>
+<TabItem value="Node.js">
+
+```javascript file=./snippets/use-secrets/raw/index.mjs
+```
+
+In this code listing:
+
+- The client's `setSecret()` method accepts two parameters - a unique name for the secret, and the secret value - and returns a new `Secret` object
+- The `withSecretVariable()` method also accepts two parameters - an environment variable name and a `Secret` object. It returns a container with the secret value assigned to the specified environment variable.
+- The environment variable can then be used in subsequent container operations - for example, in a command-line `curl` request, as shown above.
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/use-secrets/raw/main.py
+```
+
+In this code listing:
+
+- The client's `set_secret()` method accepts two parameters - a unique name for the secret, and the secret value - and returns a new `Secret` object
+- The `with_secret_variable()` method also accepts two parameters - an environment variable name and a `Secret` object. It returns a container with the secret value assigned to the specified environment variable.
+- The environment variable can then be used in subsequent container operations - for example, in a command-line `curl` request, as shown above.
+
+</TabItem>
+</Tabs>
+
+## Use secrets from the host environment
+
+Most of the time, it's neither practical nor secure to define plaintext secrets directly in your pipeline. That's why Dagger lets you read secrets from the host, either from host environment variables or from the host filesystem, or from external providers.
+
+Here's a revision of the previous example, where the secret is read from a host environment variable. To use this listing, create a host environment variable named `GH_SECRET` and assign it the value of your GitHub personal access token.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/use-secrets/host-env/main.go
+```
+
+This code listing assumes the existence of a host environment variable named `GH_SECRET` containing the secret value. It performs the following operations:
+
+- It reads the value of the host environment variable using the `Host().EnvVariable()` method.
```

Comment:
`Host().EnvVariable()` is outdated. Other languages as well.

Reply:
Thanks for catching this!

Fixed